### PR TITLE
[v7r3] Hackathon fixes

### PR DESCRIPTION
**Pull request by TaykYoku**
BEGINRELEASENOTES
Hackathon fixes

*Configuration
FIX: fix getServiceFailoverURL for multi url case

*Framework
FIX: fix getComponentsStatus bug

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py [hunk 1]
```diff
@@ -384,7 +384,8 @@ def getComponentsStatus(self, conditionDict={}):
         # Walk the URLs
         serviceURLs = getSystemURLs(system, setup)  # verify URLs in getSystemURLs method
         for service in serviceURLs:
-          url = urlparse.urlparse(serviceURLs[service])
+          # serviceURLs is a dict that contain a list of URLs for service
+          url = urlparse.urlparse(serviceURLs[service][0])
```
Comment (chaen) [inline src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py:388]: so here you arbitrarily take the first one ?

Comment (TaykYoku) [reply src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py:388]: Yes, from randomised list

**Comment by TaykYoku** [comment]
push tests

File: src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py [hunk 2]
```diff
@@ -382,9 +382,12 @@ def getComponentsStatus(self, conditionDict={}):
...
               statusSet.addUniqueToSet(requiredComponents, compDict)
         # Walk the URLs
-        serviceURLs = getSystemURLs(system, setup)  # verify URLs in getSystemURLs method
-        for service in serviceURLs:
-          url = urlparse.urlparse(serviceURLs[service])
+        systemURLs = getSystemURLs(system, setup)  # verify URLs in getSystemURLs method
+        for service in systemURLs:
+          # systemURLs is a dict that contain a list of URLs for service
+          if not systemURLs[service]:
+            return S_ERROR('Not found URL for %s service.' % service)
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py:389]: Are you sure about returning an error here? Maybe just printing it out and keep going?

File: src/DIRAC/ConfigurationSystem/Client/PathFinder.py [hunk 3]
```diff
@@ -66,22 +66,22 @@ def getSystemSection(system, instance=False, setup=False):
...
-                                'Agents', 'Services', 'Executors' or 'Databases'.
+      :param str system: system name or component name prefixed by the system in which it is placed.
+                         e.g. 'WorkloadManagement/SandboxStoreHandler'
+      :param str component: component name, e.g. 'SandboxStoreHandler'
+      :param str setup: Name of the setup.
+      :param str componentCategory: Category of the component, it can be:
+                                    'Agents', 'Services', 'Executors' or 'Databases'.
 
-  :return str: Complete path to the component
+      :return str: Complete path to the component
```
Comment (andresailer) [inline src/DIRAC/ConfigurationSystem/Client/PathFinder.py:78]:
```suggestion
      :return: Complete path to the component
      :rtype: str
```

File: src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py (hunk 2 above)
Comment (TaykYoku) [reply src/DIRAC/FrameworkSystem/DB/ComponentMonitoringDB.py]: fixed

**Review by fstagni — APPROVED** [review]

**Pull request merged**